### PR TITLE
PMM-5668 add --log-level and ENV PMM_AGENT_LOG_LEVEL

### PR DESCRIPTION
**Pull request by YaroslavPodorvanov**
PMM-5668

Build: SUBMODULES-2400

- [ ] https://github.com/percona/pmm-admin/pull/195
- [ ] https://github.com/percona/pmm-managed/pull/1035

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -146,8 +146,9 @@ type Config struct {
 	Paths  Paths  `yaml:"paths"`
 	Ports  Ports  `yaml:"ports"`
 
-	Debug bool `yaml:"debug"`
-	Trace bool `yaml:"trace"`
+	LogLevel string `yaml:"log-level"`
```
[the remark this answers is not in the document]
Comment (YaroslavPodorvanov) [reply config/config.go:149]: use "log-level" like before exists "listen-address"

File: config/logger.go [hunk 2]
```diff
@@ -42,33 +42,56 @@ var initLogger sync.Once
...
-	}
 	if logrus.GetLevel() != level {
-		logrus.Warn("Some logging settings (caller reporter, gRPC tracing) can't be changed without restart.")
+		logrus.Warn("Some logging settings (gRPC tracing) can't be changed without restart.")
+
 		logrus.SetLevel(level)
+
+		if trace {
+			// grpclog.SetLoggerV2 is not thread-safe
+			// grpclog.SetLoggerV2(&gRPCLogger{Entry: logrus.WithField("component", "grpclog")})
```
Comment (BupycHuk) [inline config/logger.go:68]: let's remove it.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: config/logger.go [hunk 3]
```diff
@@ -42,33 +42,56 @@ var initLogger sync.Once
 
 // ConfigureLogger configures standard Logrus logger.
 func ConfigureLogger(cfg *Config) {
-	initLogger.Do(func() {
-		if cfg.Debug {
-			logrus.SetLevel(logrus.DebugLevel)
-		}
+	level, trace := parseLoggerConfig(cfg.LogLevel, cfg.Debug, cfg.Trace)
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply config/logger.go:45]: I don't see any validation for possible values, so user still can use trace and will be surprised

[the remark this answers is not in the document]
Comment (YaroslavPodorvanov) [reply config/logger.go:45]: > I don't see any validation for possible values, so user still can use trace and will be surprised

we have two validation:
1 in `EnumVar(&cfg.LogLevel, "debug", "info", "warn", "error", "fatal")`
2 in `parsedLevel, err := logrus.ParseLevel(level)`

**Comment by YaroslavPodorvanov** [comment]
> we probably should pass value of log level to exporters, I don't see it in this PR.

Yes, but it will be in issue [PMM-7326](https://jira.percona.com/browse/PMM-7326), because we need pass "log-level" through pmm-managed in case when "log-level" used in pmm-admin

File: config/logger.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (BupycHuk) [reply config/logger.go:45]: 1. yeah, missed part in flag.
2. ParseLevel [supports](https://github.com/sirupsen/logrus/blob/master/logrus.go#L39) `trace`, but it doesn't matter after first point. 

anyway `trace` looks unnecessary in this context, we can just check `cfg.Trace` in that case.

**Review by ritbl — APPROVED** [review]

**Pull request merged**